code/lighting/lighting.cpp

```diff
@@ -177,12 +171,11 @@ void light_add_directional(const vec3d *dir, float intensity, float r, float g,
 	l.r = r;
 	l.g = g;
 	l.b = b;
-	l.spec_r = spec_r;
-	l.spec_g = spec_g;
-	l.spec_b = spec_b;
 
-	//Direcional lights not yet switched to use lighting profiles multipliers as that's part of a seperate project.
-	l.intensity = intensity;
+	//Direcional lights not yet switched to use lighting profiles multipliers as that's part of a seperate project
+	auto lp = lighting_profile::current();
+	l.intensity = lp->directional_light_brightness.handle(intensity);
+	l.intensity = lp->overall_brightness.handle(l.intensity);
```

Comment:
You assign intensity twice here. That seems not really correct, right?
Also, since this now _does_ use some lighting profile stuff, is the comment still accurate?

Reply:
Regarding the comment, you're correct.
Regarding the assignment, yeah, they could be done all on one line I suppose. 
```cpp
L.I = A.handle(I);
L.I= B.handle(L.I);
```
vs
```cpp
L.I = B.handle(A.handle(I));
```
I think this two line way is more plain to read and half expect the compiler to optimize away any negligible cost incurred, but I'm flexible on that if you disagree.

Reply:
Ah, I see.
IMO, we should really use the lower version, as it's much easier to tell what is actually going on. Or at least use a temporary value to hold the intermediate result from A.handle, so that it's not tripping up people reading the code.
The compiler will, as you say, optimize either to the inline version internally anyways.